src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -2018,10 +2018,10 @@ private static int FindFirstMatchedLane(Vector128<ushort> compareResult)
 
         public static void Reverse(ref char buf, nuint length)
         {
-            ref byte bufByte = ref Unsafe.As<char, byte>(ref buf);
-            nuint byteLength = length * sizeof(char);
             if (Avx2.IsSupported && (nuint)Vector256<short>.Count * 2 <= length)
             {
+                ref byte bufByte = ref Unsafe.As<char, byte>(ref buf);
+                nuint byteLength = length * sizeof(char);
                 Vector256<byte> reverseMask = Vector256.Create(
```

Comment:
The issue with when you tried to replace `Avx2.Shuffle` with `Vector256.Shuffle` is that you didn't adjust the `reverseMask`.

You should change this:
```csharp
Vector256.Create(
    (byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0,   // first 128-bit lane
          15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)); // second 128-bit lane
```

To:
```csharp
Vector256.Create(
    (byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0,   // first 128-bit lane
          31, 30, 29, 28, 27 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16)); // second 128-bit lane
```

The `Vector256` APIs operate as if it is `1x256-bit` vector rather than as `2x128-bit` vector lanes. This is consistent with how `AVX-512`, `Arm64`, `WASM`, `Vector64`, `Vector128`, and other types all operate.

Reply:
If you do this, then `Vector256.Shuffle(tempFirst, Vector256.Create(...))` will work as expected and still be performant on AVX2 hardware where you don't want to cross lanes.

Reply:
> The `Vector256` APIs operate as if it is `1x256-bit` vector

In this case, shouldn't we adjust the mask to 
```
Vector256.Create(
    (byte)31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16,
          15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0 ));
```
I noticed the issue the reverse mask but couldn't reproduce the failure yet. Does the `runfo` tool expects to use Windows? The steps using `runfo` to reproduce the pipeline failure seem create a batch file to run on Windows.

Reply:
> runfo

It can be used on Linux as well. You just need to pass the right job-id and work item which you can find it in AzDo.

<img width="1664" alt="image" src="https://user-images.githubusercontent.com/12488060/182247548-360485d9-2301-462d-9708-59da6e3d040b.png">


The `RunTests.cmd/sh` is present in the zip folder you download.

Reply:
ooh, right. Thanks Kunal.

Reply:
Can we leave the AVX2 changes out of this patch. The patch is now self contained to changes in the 128bit variants.

I'm happy for 256bit part to be changed if it's obvious, but clearly it's going to require some extra debugging to get right and not break performance (given https://github.com/dotnet/runtime/issues/72793). Let's avoid feature creeping this PR.

Reply:
I think it's fine to leave it untouched, crossplatform Vector256 apis are mostly for consistency, they're not crossplatform and unlikely to be so